torch/_tensor_docs.py

```diff
@@ -3202,10 +3238,39 @@ def callable(a, b) -> number
                r"""
 sparse_dim() -> int
 
-If :attr:`self` is a sparse COO tensor (i.e., with ``torch.sparse_coo`` layout),
+If :attr:`self` is a :ref:`sparse tensor <sparse-docs>`,
 this returns the number of sparse dimensions. Otherwise, this throws an error.
 
-See also :meth:`Tensor.dense_dim`.
+See also :meth:`Tensor.dense_dim` and :ref:`hybrid tensors <sparse-hybrid-docs>`.
+""")
+
+add_docstr_all('sparse_resize_',
+               r"""
+sparse_resize_(size, sparse_dim, dense_dim) -> Tensor
+
+Resizes :attr:`self` :ref:`sparse tensor <sparse-docs>` to the
+desired (non-shrinking) size and the number of sparse and dense
+dimensions.
+
+Args:
+    size (torch.Size): the desired size.
+    sparse_dim (int): the number of desired sparse dimensions
+    dense_dim (int): the number of desired dense dimensions
+
+If the number of specified elements of :attr:`self` tensor is 0, one
+can specify any desired size and the number of sparse/dense dimensions
+such that ``len(size) == sparse_dim + dense_dim``. Otherwise, the
+desired size must be such that any of its dimensions is greater or
+equal to the the corresponding original dimensions, and the number of
+original sparse/dense dimensions is not altered.
+""")
+
+add_docstr_all('sparse_resize_and_clear_',
+               r"""
+sparse_resize_and_clear_(size, sparse_dim, dense_dim) -> Tensor
+
+Does the same as :meth:`sparse_resize_` plus resets the `indices` and
```

Comment:
Wait what? 

Before we dig deeper into how to properly document this function, what the heck is this intended for?

Reply:
`sparse_resize_and_clear_` appears to be a worker function for `empty`, `empty_like`, `zeros`, and `zeros_like`.

Suggestion: convert it (and perhaps also `sparse_resize_`) to an internal method (read: rename to `_sparse_resize_and_clear_`).

Reply:
It turns out that this documentation string is wrong. Here's an update:
```
Removes all specified elements from a :ref:`sparse tensor
<sparse-docs>` :attr:`self` and resizes :attr:`self` to the desired
size and the number of sparse and dense dimensions.

Throws an error if :attr:`self` is not a sparse tensor.
```